src/App.jsx

```diff
@@ -14,33 +16,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABET = 'alphabet';
+const SORT_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const handleSort = (type, isReversed) => {
+  let sortedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (type) {
+    case SORT_ALPHABET:
+      sortedGoods = [...sortedGoods].sort((goods1, goods2) =>
+        goods1.localeCompare(goods2),
+      );
+      break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    case SORT_LENGTH:
+      sortedGoods = [...sortedGoods].sort(
+        (goods1, goods2) => goods1.length - goods2.length,
+      );
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setsortType] = useState('');
+  const listOfGoods = handleSort(sortType, isReversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames(`button is-info`, {
+            'is-light': sortType !== SORT_ALPHABET,
+          })}
+          onClick={() => setsortType(SORT_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames(`button is-info`, {
+            'is-light': sortType !== SORT_LENGTH,
+          })}
+          onClick={() => setsortType(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames(`button is-info`, {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortType === SORT_ALPHABET ||
+          sortType === SORT_LENGTH ||
+          isReversed) && (
```

Comment:
```suggestion
        {(sortType || isReversed) && (
```
- Just check if you have any sort, don't compare them, as in this case if you have more sort types you have to add more conditions
- and create a variable for `sortType || isReversed` to avoid combining different logical operators in the same place